mlir/lib/Analysis/FlatLinearValueConstraints.cpp

```diff
@@ -85,8 +85,10 @@ getFlattenedAffineExprs(ArrayRef<AffineExpr> exprs, unsigned numDims,
   for (auto expr : exprs) {
     if (!expr.isPureAffine())
       return failure();
-
-    flattener.walkPostOrder(expr);
+    // has poison expression
+    auto flattenResult = flattener.walkPostOrder(expr);
+    if (failed(flattenResult))
+      return failure();
```

Comment:
This is a little weird... Failure on flattening usually means it's semi-affine.  Could you update the documentation for this function?